core/src/main/scala/com/pingcap/tispark/TiDBRelation.scala

```diff
@@ -37,18 +37,44 @@ case class TiDBRelation(session: TiSession,
   @transient val sqlContext: SQLContext
 ) extends BaseRelation
     with InsertableRelation {
-  val table: TiTableInfo = meta
-    .getTable(tableRef.databaseName, tableRef.tableName)
-    .getOrElse(
-      throw new TiClientInternalException(
-        "Table not exist " + tableRef + " valid databases are: " + meta.getDatabases
-          .map(_.getName)
-          .mkString("[", ",", "]")
-      )
-    )
+  lazy val table: TiTableInfo = getTableOrThrow(tableRef.databaseName, tableRef.tableName)
 
   override lazy val schema: StructType = TiUtil.getSchemaFromTable(table)
 
+  private def getTableOrThrow(database: String, table: String): TiTableInfo =
+    meta.getTable(database, table).getOrElse {
+      val db = meta.getDatabase(database)
+      if (db.isEmpty) {
+        throw new TiClientInternalException(
+          "Database not exist " + database + " valid databases are: " + meta.getDatabases
+            .map(_.getName)
+            .mkString("[", ",", "]")
+        )
+      } else {
+        throw new TiClientInternalException(
+          "Table not exist " + tableRef + " valid tables are: " + meta
+            .getTables(db.get)
+            .map(_.getName)
+            .mkString("[", ",", "]")
+        )
+      }
+    }
+
+  lazy val isTiFlashReplicaAvailable: Boolean = {
+    // Note: INFORMATION_SCHEMA.TIFLASH_REPLICA is not present in TiKV.
```

Comment:
so where is `INFORMATION_SCHEMA.TIFLASH_REPLICA`?

Reply:
ok

Reply:
TIFLASH_REPLICA is not in TiKV or PD, it is calculated in TiDB instance and stored in memory, however it contains some info that we may want to know: progress, replica count, location lables, available.

Some of these information can be read from PD, and others may be included in TableInfo